CIP-0062/README.md

```diff
@@ -0,0 +1,220 @@
+
+This api being an extension of [CIP-30 (Cardano dApp-Wallet Web Bridge)](https://cips.cardano.org/cips/cip30/), expects that `cardano.{walletName}.enable()` to be enabled and added to CIP-30 whitelist implicitly.
+
+When both this API and **CIP-30** is being enabled, is up to the wallet to decide the number of prompts requesting permissions to be displayed to the user.
+
+# **`Jormungandr API`**
+
+## **api.submitVotes**(keyPath: KeyPath, votes: Vote[], spendingCounter: number): Promise\<hash32>
+
+`keyPath`: The derivation path values to the voting key for which transaction should be signed with. The derivation path should follow the already establish in [CIP-36 (Catalyst/Voltaire Registration Transaction Metadata Format - Updated)](https://cips.cardano.org/cips/cip36/). 
+_`m / 1694' / 1815' / account' / role' / address_index'`_
+
+### `spendingCounter`: 
+The spending counter is used to prevent double voting. The current spending counter for the account should be provided and the implementation should increment it for each vote before submission and attach it to the vote according to [Jormungandr Voting] (https://input-output-hk.github.io/jormungandr/jcli/vote.html#voting). This needs to be in sequential order.
+
+### **`votes`**: 
+
+```
+interface Vote {
+  proposal: Proposal,
+  choice: number,
+  expiration: BlockDate
+}
+```
+#### `expiration`: 
+chain epoch \& slot for when the vote will expire. The type used is: 
+
+```
+interface BlockDate {
+    epoch: number
+    slot: number
+}
+```
+
+#### `choice`: 
+The choice **index** we want to vote for. An `UnkownChoiceError` should be thrown is the value is not within the `proposal` option set.
+
+#### **`proposal`** : 
+proposal information. Include the range of options we can use to vote. This defines the allowed values in `choice`.
+
+```
+interface Proposal {
+  votePlanId: number
+  voteOptions: number[]
+}
+```
+
+##### `votePlanId`: 
+the vote plan id. This is used to identify the vote plan. This should be the same as anchored on chain.
+
+##### `voteOptions`: 
+The vote options. This is the set of options we can vote for.
+
+### **Returns**
+
+`hash32` - This is the hash of the transaction that will be submitted to the node.
+
+#### **Errors**
+`InvalidArgumentError` - Generic error for errors in the formatting of the arguments.
+
+`UnknownChoiceError` - If the `choice` is not within the `proposal` option set.
+
+`InvalidBlockDateError` - If the `validUntil` is not a valid block date.
+
+`InvalidVotingKeyError` - If the `keyPath` is not a valid voting key.
+
+`InvalidVotePlanError` - If the `votePlanId` is not a valid vote plan.
+
+`InvalidVoteOptionError` - If the `index` is not a valid vote option.
+
+# **`Delegation API`**
+
+## **api.getVotingKey**(path: KeyPath): Promise\<Bip32PublicKey>
```

Comment:
This sounds like it could be a more generic API to get _any_ public key from the wallet.

Reply:
You would need to specify what kind of key you wanted, but yes this could be extended

Reply:
@KtorZ i agree. A generic method to fetch this would warrant an update on CIP-30

Reply:
*keyPath* removed. Single Voting key generated now.